README.md

```diff
@@ -135,6 +135,28 @@ scopeUtils.satisfiesExpression(
 )
 ```
 
+If you wish to understand why a certain expression was not satisfied by a scopeset
+you can use the `explainFailure` function. The function returns a scope expression
+where all scopes that exist are missing from the scopeset. Any scopes under an
+`AllOf` key are definitely needed to satisfy the expression and at least
+one of the scopes under an `AnyOf` must be provided to satisfy.
+
+```js
+scopeUtils.explainFailure(
```

Comment:
How about calling it `scopeUtils.reduce` or `scopeUtils.simplifyScopeExpression(scopes, expression)`.

Note. I would also flip the order of the parameters.

Anyways, my suggestion is to use the terminology: _simplifying a scope expression_. Or maybe it's more of a _factor_ thing :)

Reply:
Hm, both of those names suggest that the resutling scopeExpression is equivalent to the input -- but that's not the case.  But it's also not really explaining anything.  How about unsatisfiedScopeExpression or something like that?

Reply:
How do we feel about `showMissingScopes()`?

Reply:
The order of parameters matches the one from `satisfiesExpression` which matches the old `scopeMatch` function afaict. Would you still like me to change the order for all new expressions stuff?

Reply:
> Hm, both of those names suggest that the resutling scopeExpression is equivalent to the input -- but that's not the case

true.... well, they are equivalent under the assumption that the given scopes are satisfied.
So yes, simplify might be a bad term... reduce is slightly better...

> How do we feel about `showMissingScopes()`?

it doesn't convey that the result is also a valid _scope-expression_ or `null`.

How about: `reduceGiven(scopes, expr)` or `reduceExpression` or `removeScopes(scopes, expr)` or `factorOutScopes(scopes, expr)`... Since this is really about factoring out a set of scopes that is given. One could also call it `reduceScope(scopes , expr)` since we're reducing the scope of the expression (I don't like this name).

Or just `removeGivenScopes(scopes, expr)` or `reduceGivenScopes(scopes, expr)`?

-----

> Would you still like me to change the order for all new expressions stuff?

No, it was just a nit... it's not important..